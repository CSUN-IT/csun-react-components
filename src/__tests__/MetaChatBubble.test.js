import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import shortid from 'shortid';
import MetaChatBubble from '../../src/components/web/ChatBubble';

configure({ adapter: new Adapter() });

describe('MetaChatBubble', () => {
  it('Should be defined', () => {
    expect(MetaChatBubble).toBeDefined();
  });

  it('Renders component correctly', () => {
    shortid.generate = jest.fn(() => 0);

    const tree = shallow(
      <MetaChatBubble
        className={'test'}
        text="Hey I'm Matty!\nHow can I help you"
        subText="This will be small"
        receiver
      />,
      <MetaChatBubble
        text="Hey I'm sending you this"
        subText="This will be small too."
        receiver={false}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
