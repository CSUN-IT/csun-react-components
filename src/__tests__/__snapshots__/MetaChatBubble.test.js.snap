// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaChatBubble Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaChatBubble
    className="test"
    receiver={true}
    subText="This will be small"
    text="Hey I'm Matty!\\\\nHow can I help you"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="bubble receiver"
      >
        <p>
          Hey I'm Matty!
        </p>
        <p>
          How can I help you
        </p>
      </div>,
      "className": "test bubble-container bubble-container--left",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Hey I'm Matty!
          </p>,
          <p>
            How can I help you
          </p>,
        ],
        "className": "bubble receiver",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Hey I'm Matty!",
          },
          "ref": null,
          "rendered": "Hey I'm Matty!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "How can I help you",
          },
          "ref": null,
          "rendered": "How can I help you",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="bubble receiver"
        >
          <p>
            Hey I'm Matty!
          </p>
          <p>
            How can I help you
          </p>
        </div>,
        "className": "test bubble-container bubble-container--left",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Hey I'm Matty!
            </p>,
            <p>
              How can I help you
            </p>,
          ],
          "className": "bubble receiver",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "Hey I'm Matty!",
            },
            "ref": null,
            "rendered": "Hey I'm Matty!",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "How can I help you",
            },
            "ref": null,
            "rendered": "How can I help you",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): <MetaChatBubble
    className=""
    receiver={false}
    subText="This will be small too."
    text="Hey I'm sending you this"
  />,
}
`;
